frontend/lib/src/components/elements/DeckGlJsonChart/DeckGlJsonChart.tsx

```diff
@@ -144,35 +151,51 @@ export class DeckGlJsonChart extends PureComponent<PropsWithHeight, State> {
     return null
   }
 
-  static getDeckObject = (props: PropsWithHeight): DeckObject => {
-    const { element, width, height, theme } = props
-    const json = JSON.parse(element.json)
+  static getDeckObject = (
+    props: PropsWithHeight,
+    state: Partial<State>
+  ): DeckObject => {
+    const { element, width, height, theme, isFullScreen } = props
+    const { id } = state
+
+    const currFullScreen = isFullScreen ?? false
+
+    // Only parse JSON when not transitioning to/from fullscreen, the element id changes, or theme changes
+    if (
+      element.id !== id ||
+      state.isFullScreen !== currFullScreen ||
```

Comment:
Why do we need to reparse it when fullscreen changes and not when `width` changes?

Reply:
Good point. Will test that.

Reply:
<img width="1107" alt="Screenshot 2023-08-07 at 3 26 41 PM" src="https://github.com/streamlit/streamlit/assets/103006371/c103b6ab-22f4-4eab-ae4c-c634764f7b6a">
<img width="1665" alt="Screenshot 2023-08-07 at 3 26 35 PM" src="https://github.com/streamlit/streamlit/assets/103006371/1f7f9f19-ce79-43e5-b9cc-03de0010163e">
Looks like it's fine!